# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/generic/DocAccessible.cpp
# Commit: 4c518f1d119a
# Full Hash: 4c518f1d119aad62e634548f9281e3d803a924b2
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2020-02-21 05:56:31
# Description:
#   Bug 1616797 - Check that accessible's frame still exists in scroll dispatcher r=Jamie
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D63546
# ==============================================================================

diff -r a0b503d3b39d -r 4c518f1d119a accessible/generic/DocAccessible.cpp
--- a/accessible/generic/DocAccessible.cpp	Thu Feb 20 20:55:23 2020 +0000
+++ b/accessible/generic/DocAccessible.cpp	Thu Feb 20 21:12:43 2020 +0000
@@ -2531,11 +2531,18 @@
     return;
   }
 
+  nsIFrame* frame = acc->GetFrame();
+  if (!frame) {
+    // Although the accessible had a frame at scroll time, it may now be gone
+    // because of display: contents.
+    return;
+  }
+
   LayoutDevicePoint scrollPoint;
   LayoutDeviceRect scrollRange;
   nsIScrollableFrame* sf = acc == this
                                ? mPresShell->GetRootScrollFrameAsScrollable()
-                               : acc->GetFrame()->GetScrollTargetFrame();
+                               : frame->GetScrollTargetFrame();
 
   // If there is no scrollable frame, it's likely a scroll in a popup, like
   // <select>. Just send an event with no scroll info. The scroll info
